Emby.Dlna/PlayTo/DlnaHttpClient.cs

```diff
@@ -54,15 +56,34 @@ private static string NormalizeServiceUrl(string baseUrl, string serviceUrl)
                     LoadOptions.None,
                     cancellationToken).ConfigureAwait(false);
             }
-            catch (XmlException ex)
+            catch (XmlException)
             {
-                _logger.LogError(ex, "Failed to parse response");
-                if (_logger.IsEnabled(LogLevel.Debug))
+                // try correcting the Xml response with common errors
+                var xmlString = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
+
+                // find and replace unescaped ampersands (&)
+                Regex regex = new Regex(@"(&(?![a-z]*;))");
+                xmlString = regex.Replace(xmlString, @"&amp;");
+
+                try
                 {
-                    _logger.LogDebug("Malformed response: {Content}\n", await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false));
+                    // retry reading Xml
+                    var xmlReader = new StringReader(xmlString);
+                    return await XDocument.LoadAsync(
+                        xmlReader,
+                        LoadOptions.None,
+                        cancellationToken).ConfigureAwait(false);
                 }
+                catch (XmlException ex)
+                {
+                    _logger.LogError(ex, "Failed to parse response");
+                    if (_logger.IsEnabled(LogLevel.Debug))
```

Comment:
Serilog already does this

Reply:
Just a copy of what was there already, but I'm all for readability!

Reply:
The original intention of this was to only execute the expensive ReadAsString method when it would actually be logged